### ISSUE TITLE
editing an existing commit changes the author of the commit

### ISSUE THREAD

**Issue by CheyenneWills** [sublimehq/sublime_merge#728]
**Version info**

 - OS: Gentoo Linux
 - Build: sublime merge Dev Channel Build 2006 (but happens on stable as well)

**Description**

When editing the contents of an existing commit the author field gets replaced with the current "user" information.

**Steps to reproduce**

1. From an existing commit where the author is not "you" Select "Edit Commit Contents" from the context menu
2. make some modification, "stage" the change, "commit" the change and "continue rebase" 

**Expected behavior**

The original Author be altered.  (Or at least make it an option to change/not change the author).

**Comment by CheyenneWills** [comment]
I'm poking at this issue mainly because the workaround for this issue is the most disruptive item in my git workflow.

The project that I do most of my work with requires that the original author commits be maintained when they are back-ported into earlier releases. (These could be bug fixes or selected new features which could contain a large number of commits).

Thanks in advance

**Comment by srbs** [comment]
My recommendation is to just use `git rebase -i` on the command line to edit commits you need to preserve authorship of.

**Comment by CheyenneWills** [comment]
Yes, that's the workaround.  But the whole point of having a tool like sublime_merge is to not be dropping in to the command line.  So this particular issue makes SM almost useless when doing certain stages of back porting features because I end up just working from the command line.

**Comment by srbs** [comment]
@CheyenneWills, I have a workaround for you:

requirements:
* `perl`

1. add this git alias:
```bash
git config --global alias.smerge-edit-commit-contents '!f() {
    local commitish target
    
    commitish="$(git rev-parse --verify --quiet --short --end-of-options "${1:-HEAD}")"
    if [ -z "$commitish" ]; then
        echo "Error: commit hash cannot be identified for $1" >&2
        exit 1
    fi
    
    # root commit check
    target="$commitish~"
    if ! git rev-parse --verify --quiet $target; then
        target="--root"
    fi
    
    # change rebase editor to perl to regex-replace & not edit
    # note: perl for simple multi-platform support
    # note: `(# )?` is for root commit support
    GIT_SEQUENCE_EDITOR="perl -p -i -e '"'"'s/(# )?pick $commitish/edit $commitish/'"'"'" \
            git rebase -i "$target"
}; f'
```
2. add this menu item to `Commit.sublime-menu`:
```json
[
    {
        "id": "edit",
        "children":
        [
            {
                "caption": "Edit Commit Contents (NO AUTHOR CHANGE)",
                "command": "git",
                "args": { "argv": ["smerge-edit-commit-contents", "$commit"] },
            }
        ]
    }
]
```

**Comment by CheyenneWills** [comment]
Thanks 

I'll play around with it.

**Comment by CheyenneWills** [comment]
It seems to be a workable work around.  Just a slightly different set of steps to make the changes (e.g. stage then amend previous commit)

**Comment by CheyenneWills** [comment]
Just a bit of reason for retaining the original author of the commit.  

In the one project where I do a lot of git work, the policy of the project is to retain the original author when a commit is "pulled up" from the main development branch into the stable branch(es), which may require some minor changes to the commit to address differences between the main branch and the individual stable branches (we use git cherry-pick -x when pulling up the commit(s)).

The project is in the process of changing the license used within the project, which means we need to contact all the authors of all the commits, so retaining the original authors match up is helpful when there was some rework required for a pull up.
